sdk/src/ed25519_instruction.rs

```diff
@@ -29,9 +29,9 @@ pub struct Ed25519SignatureOffsets {
     message_instruction_index: u16,    // index of instruction data to get message data
 }
 
-pub fn new_ed25519_instruction(keypair: &ed25519_dalek::Keypair, message: &[u8]) -> Instruction {
+pub fn new_ed25519_instruction(keypair: &ed25519_dalek::SigningKey, message: &[u8]) -> Instruction {
```

Comment:
This might be a backward incompatible change as well.
Even if `SigningKey` is the same as `Keypair`, previous version was using `ed25519_dalek` version `1.0.1`, and it would be considered compatible with the same time from `2.x`, if version dependencies are using the default SemVer constraints.